### PR TITLE
clean-up: added aliases to prevent 404 of old paths, and content clean up

### LINKED ISSUE #628: could it be possible we make this path transition seamlessly?
> Related to this PR: maybe consider redirecting old doc links to the new doc URLs at least until search engines re-index (or redirect all `/en/docs/*` to documents home if that is difficult). Currently if you search Google for Gobuffalo docs there are a lot of 404s

from https://github.com/gobuffalo/docs/pull/627#issuecomment-1088193599 by @saurori 

Yeah, I agree. Actually, I am not a Hugo person but a Jekyll lover, and Jekyll has a plugin called `jekyll-redirect-from` which supports a directive called `redirect_from` for front-matter. With this directive and document template with the `canonical` link for SEO, seamless moving was possible when I moved my blog.

I quickly search for it, and I found there was a discussion about it before. [1] Do you know how `aliases` directive works? I think it could be a solution for this issue.

[1] https://discourse.gohugo.io/t/url-redirect-forwarding/4689/19

### PR DESCRIPTION
**Pull request by sio4**
Originally started to add aliases but...
- added aliases to prevent 404 when people click on the web search. (however, many of them are already updated)
- changed some filename to make the path string closer to its content
- fixed links to the other pages
- restore missing example code (but... readability is not yet good)
- titlelized or added missing headers
- look and feel tweak: added space between header leading blocks
- look and feel tweak: changed color and size of header anchors
- fixed some confirmed external links

fixes #628

### CODE REVIEW

**Comment by sio4** [comment]
Hi @paganotoni 
This is a PR for #628. Please take a look at this.

File: assets/css/home.css [hunk 1]
```diff
@@ -35,14 +35,17 @@ html {
 }
 .content h1 {
     @apply text-2xl md:text-4xl font-bold mb-3;
+		margin-bottom: 1.2em;
```
Comment (paganotoni) [inline assets/css/home.css:38]: As we're using tailwindcss. Lets change this one to be `@apply mb-1` or mb1.5 depending of what works best.

**Review by paganotoni — CHANGES_REQUESTED** [review]
Some recommendations to keep close to tailwindcss. Thanks @sio4

File: assets/css/home.css (hunk 1 above)
Comment (sio4) [reply assets/css/home.css]: Ah, I heard one of my colleagues talk about tailwind something! maybe this? committed!

**Review by paganotoni — APPROVED** [review]
THanks!

**Pull request merged**